Omnition/omnition-opentelemetry-collector
# Simple port of SignalFx agent's memory monitor

JIRA: A0-1888

TESTING: Locally with SignalFx agent and SignalFx dashboard

A test PR trying to port a single SignalFx monitor to Otel as a receiver. Some initial thoughts:

#### Notes

- Simple monitors should not take more than a one day to port and be production ready. More complicated ones might take longer.
- Otel does not support events yet which are used by some monitors.
- Otel converts dots to underscore. For example, it converts memory.used to memory_user. Need to investigate whether it's Otel (OC metrics) or prometheus metrics that make the conversion.
- SignalFx dashboard by default applied per second rate rollup to metrics exported by Otel. Need to investigate why. Probably becuase SignalFx backend has some well known metrics and it knows the correct rollup to apply. For example it understands memory.used should have average rollup but not memory_used (which otel exports)
- Ben is working on a document to list down all features Otel is missing that SignalFx monitors need.

#### Testing

- Ported a simple (memory) monitor from SignalFx to Otel as a receiver.
- Setup a metrics pipeline in Otel with prometheus exporter making the metrics available for consumption.
- Setup two monitors in SignalFx agent:
  - Memory monitor with status label (source="signalfx-agent")
  - PrometheusExporter monitor
    - This monitor is configured to fetch metrics exposed by Otel and forward them to SignalFx backend.
- Compare metrics in SignalFx backend from the two sources and verify correctness.


![image](https://user-images.githubusercontent.com/46186/68023768-3b4ee700-fcce-11e9-8247-5f6316a8a6cc.png)

Thanks for all the help setting up SignalFx agent and backend @keitwb
